_posts/2019-03-18-legacy-rails-apps-silently-judging-you.md

```diff
@@ -0,0 +1,236 @@
+---
+layout: post
+title:  "Legacy Rails: Silently Judging You"
+date: 2019-03-18 10:00:00
+categories: ["rails", "upgrades"]
+author: "etagwerker"
+---
+
+I had to come up with a clever title because this article is about legacy
+[Rails](https://rubyonrails.org/) applications and I know that you might fall
+asleep by the third paragraph. **Boooooring...** You probably want to read about
+that new JavaScript framework that came out (_I love that this sentence will always be true, it doesn't matter when you read this_)
+
+If you have been working with Rails for a few years, you have seen your
+fair share of shiny new applications, well-maintained and poorly-maintained
+legacy applications. This post is about **Legacy Rails applications**
+
+So let's start judging them! _Are they in good shape? Should we upgrade them to
+a more recent version of Rails? Should we re-write them?_
+
+<img src="/blog/assets/images/judging.gif" alt="Judging You" class="half-img">
+
+<!--more-->
+
+How can you judge the quality of a Rails application? Here is a process that
+works for us.
+
+## Dependency Archeology
+
+A good way to get started is to find out _how_ outdated an application really is.
+You can easily do this with [Bunder](https://bundler.io/)'s `bundle outdated`:
+
+```
+$ bundle outdated
+Fetching gem metadata from https://rubygems.org/..........
+Resolving dependencies.......................................................................................................................................................
+
+Outdated gems included in the bundle:
+  * actionmailer (newest 5.2.2, installed 3.2.22.5)
+  * actionpack (newest 5.2.2, installed 3.2.22.5)
+  * activemerchant (newest 1.90.0, installed 1.53.0) in groups "default"
+  * activemodel (newest 5.2.2, installed 3.2.22.5)
+  ...
+  * tilt (newest 2.0.9, installed 1.4.1)
+  * todonotes (newest 0.2.2, installed 0.1.0)
+  * treetop (newest 1.6.10, installed 1.4.15)
+  * tzinfo (newest 2.0.0, installed 0.3.52)
+  * uglifier (newest 4.1.20, installed 1.2.4) in groups "assets"
+  * unicorn (newest 5.4.1, installed 4.8.3) in groups "default"
+  * uniform_notifier (newest 1.12.1, installed 1.4.0)
+  * whenever (newest 0.10.0, installed 0.8.2) in groups "default"
+  * wicked_pdf (newest 1.1.0, installed 0.9.10) in groups "default"
+```
+
+Using this information, you can review the `Gemfile` and improve it. If you find
+dependencies that are declared like this:
+
+```ruby
+gem 'rails', '4.2.7'
+```
+
+You could improve them like this:
+
+```ruby
+gem 'rails', '~> 4.2.7'
+```
+
+Next time you run `bundler update` it will jump to the latest patch version.
```

Comment:
A brief explanation of _why_ this will jump to the latest patch version would be useful here.